core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -149,20 +169,24 @@ private OptionalLong lockBlockInternal(long sessionId, long blockId, BlockLockTy
         return OptionalLong.empty();
       }
     }
+    long lockId = LOCK_ID_GEN.getAndIncrement();
+    LockRecord record = new LockRecord(sessionId, blockId, lockId, lock);
     try {
-      long lockId = LOCK_ID_GEN.getAndIncrement();
-      try (LockResource r = new LockResource(mSharedMapsLock.writeLock())) {
-        mLockIdToRecordMap.put(lockId, new LockRecord(sessionId, blockId, lock));
-        Set<Long> sessionLockIds = mSessionIdToLockIdsMap.get(sessionId);
-        if (sessionLockIds == null) {
-          mSessionIdToLockIdsMap.put(sessionId, Sets.newHashSet(lockId));
-        } else {
-          sessionLockIds.add(lockId);
-        }
+      try {
+        mSessionCleaning.acquire(0);
```

Comment:
I'm a little bit confused here about acquire 0. acquire(0) would not block if it's 0 semaphore. We can discuss offline about this design
```
    Semaphore test = new Semaphore(0);
    test.acquire(0);
    System.out.println("test");
```

Reply:
Good catch! Indeed the semaphore need to acquire at least 1 permit to be able to potentially block. Changed this to an effectively unlimited number of permits.